status/condition_set.go

```diff
@@ -54,9 +54,16 @@ type ConditionSet struct {
 func (r ConditionTypes) For(object Object) ConditionSet {
 	cs := ConditionSet{object: object, ConditionTypes: r}
 	// Set known conditions Unknown if not set.
-	for _, t := range append(r.dependents, r.root) {
+	// Set the root condition first to get consistent timing for LastTransitionTime
+	for _, t := range append([]string{r.root}, r.dependents...) {
 		if cs.Get(t) == nil {
-			cs.SetUnknown(t)
+			cs.Set(Condition{
+				Type:               t,
+				Status:             metav1.ConditionUnknown,
+				Reason:             "AwaitingReconciliation",
+				Message:            "object is awaiting reconciliation",
+				LastTransitionTime: object.GetCreationTimestamp(),
```

Comment:
Optional: just call SetUnknown, and make it set creation timestamp if not currently set.

Reply:
> and make it set creation timestamp if not currently set

How do you do that? You don't access to the creation timestamp since SetUnknown is on the condition set. You need some way to pass-down that detail

Reply:
Ah, didn't realize that the ConditionSet had access to the object -- should have actually looked at the struct